kartothek/io/dask/delayed.py

```diff
@@ -70,6 +70,10 @@ def delete_dataset__delayed(dataset_uuid=None, store=None, factory=None):
         load_dataset_metadata=False,
     )
 
+    garbage_collect_dataset__delayed(
```

Comment:
This way, the GC is never executed. The `garbage_collect_dataset__delayed` only returns a delayed object which needs to be dispatched to a dask scheduler. See the usage of `delayed_dataset_uuid` and `mps` below

Reply:
so we need to use `delayed_dataset_uuid` and `mps` in garbage_collect_dataset__delayed ?

Reply:
The return value of `garbage_collect_dataset__delayed` is a `dask.delayed` object and needs to be built into the graph. The `_delete_tl_metadata` function is built as a _collector_, i.e. it accepts multiple arguments. By passing a delayed object to this function, the delayed object is calculated befor that function.

```
gc = garbace_collect_dataset__delayed(...)
mps = map_delayed(...)
delayed(_delete_tl_metadata)(dataset_factory, mps, gc)
```

Reply:
Push the latest changes